src/translator/service.cpp

```diff
@@ -28,10 +28,10 @@ loadVocabularies(marian::Ptr<marian::Options> options) {
 namespace marian {
 namespace bergamot {
 
-Service::Service(Ptr<Options> options, const void *model_memory)
+Service::Service(Ptr<Options> options, MemoryGift modelMemory, MemoryGift shortlistMemory)
     : requestId_(0), vocabs_(std::move(loadVocabularies(options))),
       text_processor_(vocabs_, options), batcher_(options),
-      numWorkers_(options->get<int>("cpu-threads")), model_memory_(model_memory)
+      numWorkers_(options->get<int>("cpu-threads")), modelMemory_(&modelMemory),shortlistMemory_(&shortlistMemory)
```

Comment:
`MemoryGift` is taken by value, does it even live past the constructor? Isn't this a potential issue?

Reply:
Shall I pass in MemoryGift pointers instead?

Reply:
I'm in favour of pointers for the scope of this PR. The contents referred to by this pointer should not be modified by `Service` or `BatchTranslator` (have compiler complain if attempted to by some const pointer mechanism). This makes it clear from types that we do not own this memory.

Reply:
OK. Will switch to `const MemoryGift *`.